dll/win32/kernel32/kernel32_vista/vista.c

```diff
@@ -26,19 +26,37 @@
  */
 BOOL
 WINAPI
-QueryFullProcessImageNameW(HANDLE hProcess,
-                           DWORD dwFlags,
-                           LPWSTR lpExeName,
-                           PDWORD pdwSize)
+QueryFullProcessImageNameW(
+    _In_ HANDLE hProcess,
+    _In_ DWORD dwFlags,
+    _Out_ LPWSTR lpExeName,
```

Comment:
Candidates for a better SAL annotation:
```suggestion
    _Out_writes_to_(*pdwSize, *pdwSize + 1) _Post_z_
        LPWSTR lpExeName,
```
or alternatively,
```suggestion
    _Out_z_cap_post_count_(*pdwSize, *pdwSize + 1)
        LPWSTR lpExeName,
```
Whichever chosen one can be also used for the A-version of the function.

Reply:
I'd prefer basic annotation with documentation rather than long annotation.

Reply:
The annotations are for the static analyzers to verify how the function uses its parameters, and how its callers use them too. Therefore the annotations better be precise.

Reply:
> The annotations are for the static analyzers

OK.